birdhouse/components/weaver/.gitignore

```diff
@@ -1,4 +1,5 @@
-conf.extra-service.d/weaver.conf
```

Comment:
BLOCKING: Keeping all old path in `.gitignore` and old `.gitignore` file if deleted otherwise this will break autodeploy hence breaking backward compat.

Reply:
We wouldn't want to delete `env.local` either so we could do:

```sh
mv env.local env.local.backup
git clean -Xf -- .
mv env.local.backup env.local
```

Reply:
> @tlvu Having to keep all old entries in the gitignore file will build up over time and could get very messy.

@mishaschwartz Yes but it make the PR 100% back compat, meaning zero manual intervention when this PR go live via autodeploy unattended deployment. Also no need to take special care of `env.local` either to avoid it being lost.

Note we are at least 3 orgs (Ouranos, CRIM, PCIC) and each org can have more than 1 deployments.  The number of manual interventions saved is significant.

Yes this could get messy with time but we do not rename or move config files that often so I think the benefits outweigh the risks.

Reply:
@tlvu I'll go with your strategy for this PR and keep the old gitignore entries. But....

> The number of manual interventions saved is significant.

If we add the git clean command to `triggerdeploy.sh` then the cleanup is automatic and no manual intervention is needed, and the PR remains backwards compatible. I'll make an issue about it once this PR is pulled in so we don't forget to remove them eventually.

Reply:
> If we add the git clean command to `triggerdeploy.sh` then the cleanup is automatic and no manual intervention is needed, and the PR remains backwards compatible. I'll make a PR for it once this PR is pulled in so we don't forget to remove them eventually.

Please make this optional, activated via a `env.local` config, and not activated by default.

Sometime we need to investigate issues on prod and need manual change to those config files or other files that are not config files.  We do not want these files to be crushed on an autodeploy.  There might be other files like `env.local` that is ignored but should not be wiped.

Any production deployment activating this option should be fully aware of the possible risks of losing unsaved changes.  Put a warning comment in `env.local.example`.

On dev machine, this should probably never be turned on to avoid interfering with debugging process.

Reply:
Thinking about this, I think a safer way to perform unattended cleanup these old template expanded files is to have a hardcoded list of those old names and delete them.

So any renamed or deleted template expanded file path is added to that list and wipe.  This way we guaranty to only clean up template expanded file and not accidentally delete something else.

Someone later can add another ignore file that should not be wiped (ex: `env.local`) and if he or she forgot to add to some sort of wipe exclude list, that file will be wiped by `git clean -Xf`.

Any unattended deletion, we better be safe than sorry.

Reply:
Hum, in fact, all external components can perform this unattended cleanup of old template expanded files in their respective post-compose script because autodeploy breakage due to those files only happen at the subsequent autodeploy run.  

So this feature can be applied for all external repos, no need for any additional "framework/platform support".

Thanks for making me think about this.